### PR TITLE
A CDM user can create a new TradeLot with a unique Identifier

### PR DESCRIPTION
**Pull request by JayasriR**

### CODE REVIEW

File: rosetta-source/src/main/resources/cdm-sample-files/functions/business-event/quantity-change/increase-equity-swap-existing-trade-lot-func.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+Increase Business Event with Existing Trade Lot
+
+New Trade
+
+- New Equity execution is performed on `2018-04-01` between parties `LEI1RPT0001` and `LEI2CP0002`.
+- Notional of `28,469,376 USD`.
```
Comment (hugohills-regnosys) [inline rosetta-source/src/main/resources/cdm-sample-files/functions/business-event/quantity-change/increase-equity-swap-existing-trade-lot-func.md:6]: You can remove this line as it is repeated below

File: rosetta-source/src/main/resources/cdm-sample-files/functions/business-event/quantity-change/partial-termination-equity-swap-func.md [hunk 2]
```diff
@@ -5,13 +5,17 @@ Equity Swap
...
+  - Notional of `35,969,376 USD`
+  - `1010400` shares with price of `67.44 USD`
 
 Partial Termination
 
-- A Partial Termination is performed on `2021-11-11`. 
-  - `760,400` Shares are closed, reducing the Quantity and Notional on the TradeLot `LOT-1` to `0`.
-  - Cash price of `2,000 USD`.
\ No newline at end of file
+- A Partial Termination is performed on `2021-11-11`. `202,080` Shares are closed
```
Comment (hugohills-regnosys) [inline rosetta-source/src/main/resources/cdm-sample-files/functions/business-event/quantity-change/partial-termination-equity-swap-func.md:17]: A Partial Termination is performed on `2021-11-11`.  The notional is decreased by `xxx` and the number of shares is decreased by `xxx`

File: RELEASE.md [hunk 3]
```diff
@@ -1,25 +1,19 @@
-# *Eligible Collateral Representation - New Attributes*
+# *CDM support for increase on existing Trade Lot*
```
Comment (hugohills-regnosys) [inline RELEASE.md:1]: `Product Model: Quantity Change For Existing TradeLot`

File: RELEASE.md [hunk 4]
```diff
@@ -1,25 +1,19 @@
...
 
 _Background_
 
-Through the Collateral Working Group, members have requested two enhancements to the modelling of collateral eligibility to enhance the CDM's capability to support
-additional use cases:
-1. To support the scenario where a legacy collateral schedule has shared criteria for IM and VM, with selected terms applicable to only one single margin type
-2. To prioritise between Collateral Criteria where agency ratings are specified for both assets and issuers
+CDM does not allow for an existing tradeLot to be increased by a delta amount, however it does allow decrease via a delta amount.
```
Comment (hugohills-regnosys) [inline RELEASE.md:5]: This release adds support for price quantity changes on an existing `TradeLot`, as described in issue [#2923](https://github.com/finos/common-domain-model/issues/2923).

File: RELEASE.md [hunk 5]
```diff
@@ -1,25 +1,19 @@
...
-- Two new attributes added to `CollateralCriteriaBase` to increase the specificity of the definition of the criteria in which collateral is eligible:
-    - `restrictTo` to denote whether the criteria applies to only IM or VM, using the existing enumerator `CollateralMarginTypeEnum`
-    - `ratingPriorityResolution` to denote whether the Issuer Criteria or Asset Criteria have precedence where there are multiple Agency Ratings defined,
-  using the new `RatingPriorityResolutionEnum` enumerator
 
-Both new attributes are optional, with singular cardinality, so this is a backward-compatible change. 
+The `Create_Quantity_Change` function has been updated to enable the user to either create a new trade lot by populating a new lot identifier for any prior tradeLot not linked to the parent trade, 
+or, if the tradeLot identifier on an increase or decrease matches an existing tradeLot identifier, to apply the quantityChange to the existing tradeLot.
 
 _Review Directions_
```
Comment (hugohills-regnosys) [inline RELEASE.md:15]: Need to add details here about the updated sample files (viewable in github), and in Rosetta going to the visualisations tab

File: rosetta-source/src/main/resources/cdm-sample-files/functions/business-event/quantity-change/increase-equity-swap-func-input.json [hunk 6]
```diff
@@ -3,6 +3,26 @@
     "primitiveInstruction" : {
```
Comment (hugohills-regnosys) [inline rosetta-source/src/main/resources/cdm-sample-files/functions/business-event/quantity-change/increase-equity-swap-func-input.json:3]: Are these changes expected?  I thought the original increase use case was unchanged?

File: rosetta-source/src/test/java/org/isda/cdm/functions/FunctionInputCreationTest.java [hunk 7]
```diff
@@ -415,22 +418,104 @@ void validatePartialTerminationEquitySwapFuncInputJson() throws IOException {
...
+        tradableProductBuilder.addTradeLot(tradeLot2Builder);
+        return tradeStateBuilder.build();
+    }
+
     /**
      * This is in a separate method because it is used by validateIncreaseEquitySwapFuncInputJson (to validate input),
      * and validatePartialTerminationEquitySwapFuncInputJson (as the input uses the output of the increase func).
      */
 
     private CreateBusinessEventInput getIncreaseEquitySwapFuncInputJson() throws IOException {
```
Comment (hugohills-regnosys) [inline rosetta-source/src/test/java/org/isda/cdm/functions/FunctionInputCreationTest.java:512]: I would not expect changes to the original increase sample

File: rosetta-source/src/test/java/org/isda/cdm/functions/FunctionInputCreationTest.java [hunk 8]
```diff
@@ -385,25 +385,28 @@ void validatePartialTerminationVanillaSwapFuncInputJson() throws IOException {
     @Test
     void validatePartialTerminationEquitySwapFuncInputJson() throws IOException {
         // The tradeState input to partial termination is output from increase event
-        CreateBusinessEventInput increaseEquitySwapInput = getIncreaseEquitySwapFuncInputJson();
+        CreateBusinessEventInput increaseEquitySwapInput = getIncreaseEquitySwapExistingTradeLotFuncInputJson();
         BusinessEvent increaseOutput = runCreateBusinessEventFunc(increaseEquitySwapInput);
         TradeState increaseTradeState = increaseOutput.getAfter().get(0);
```
Comment (hugohills-regnosys) [inline rosetta-source/src/test/java/org/isda/cdm/functions/FunctionInputCreationTest.java:390]: This needs looking at.. 
- The original code gets the increase equity swap func input (which has 1 tradelot), and actually runs the function so a 2nd tradeLot is added. It then extracts the output (which has 2 tradeLots).. the output is used as the input to the partial termination example.
- The new code gets the increase equity swap with existing trade lot func input (which already has 2 tradeLots), so it's not really necessary to then run the function

**Review by eacunaISDA — DISMISSED** [review]

**Review by hugohills-regnosys — APPROVED** [review]

**Pull request merged**